src/layout/Button/ButtonComponent.tsx

```diff
@@ -46,41 +55,50 @@ export const ButtonComponent = ({ node, ...componentProps }: IButtonReceivedProp
   };
 
   const submitTask = ({ componentId }: { componentId: string }) => {
-    const { org, app, instanceId } = window as Window as IAltinnWindow;
-    if (currentTaskType === 'data') {
-      dispatch(
-        FormDataActions.submit({
-          url: `${window.location.origin}/${org}/${app}/api/${instanceId}`,
-          apiMode: 'Complete',
-          stopWithWarnings: false,
-          componentId,
-        }),
-      );
-    } else {
-      dispatch(ProcessActions.complete());
+    if (!disabled) {
+      const { org, app, instanceId } = window as Window as IAltinnWindow;
+      if (currentTaskType === 'data') {
+        dispatch(
+          FormDataActions.submit({
+            url: `${window.location.origin}/${org}/${app}/api/${instanceId}`,
+            apiMode: 'Complete',
+            stopWithWarnings: false,
+            componentId,
+          }),
+        );
+      } else {
+        if (processActionsFeature) {
+          dispatch(ProcessActions.complete({ action: 'confirm' }));
+        } else {
+          dispatch(ProcessActions.complete());
+        }
+      }
```

Comment:
Maybe we should check the task type here as well? If not, suddenly a new task type is introduced by someone, we forget about this code snippet, and suddenly the 'confirm' action is sent on an entirely different task type.. 🤔 If I understood this correctly, that is. Might be better to have a fallback to just complete the process task without setting an action.

Reply:
good point